### ISSUE TITLE
Unable to uninstall winget itself.

### ISSUE THREAD

**Issue by fxzxmicah** [microsoft/winget-cli#2648]
### Brief description of your issue

I cannot uninstall winget because of this property: `NonRemovable: True`.
I think that in many people, including me, winget or Appinstaller is not a necessary program. We can use PowerShell to install programs, or simply do not use Appx based programs.

### Steps to reproduce

Execute command in PowerShell: `Remove-AppPackage Microsoft.DesktopAppInstaller_1.18.2691.0_x64__8wekyb3d8bbwe`

### Expected behavior

Successfully uninstalled the program.

### Actual behavior

Throw an error: `0x80073CFA`

### Environment

```shell
Name              : Microsoft.DesktopAppInstaller
Publisher         : CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US
Architecture      : X64
ResourceId        :
Version           : 1.18.2691.0
PackageFullName   : Microsoft.DesktopAppInstaller_1.18.2691.0_x64__8wekyb3d8bbwe
InstallLocation   : C:\Program Files\WindowsApps\Microsoft.DesktopAppInstaller_1.18.2691.0_x64__8wekyb3d8bbwe
IsFramework       : False
PackageFamilyName : Microsoft.DesktopAppInstaller_8wekyb3d8bbwe
PublisherId       : 8wekyb3d8bbwe
IsResourcePackage : False
IsBundle          : False
IsDevelopmentMode : False
NonRemovable      : True
Dependencies      : {Microsoft.UI.Xaml.2.7_7.2208.15002.0_x64__8wekyb3d8bbwe, Microsoft.VCLibs.140.00.UWPDesktop_14.0.3
                    0704.0_x64__8wekyb3d8bbwe, Microsoft.DesktopAppInstaller_1.18.2691.0_neutral_split.language-zh-hans
                    _8wekyb3d8bbwe, Microsoft.DesktopAppInstaller_1.18.2691.0_neutral_split.scale-125_8wekyb3d8bbwe}
IsPartiallyStaged : False
SignatureKind     : Store
Status            : Ok
```

**Comment by fxzxmicah** [comment]
> Winget has nothing to do with Appx-based programs. 

First of all, I need to make it clear that currently Winget is bound to Windows Appinstaller, which is used to install appx-based programs, so it is indeed related to appx-based programs.

> it simplifies the process of finding a program from developer website, downloading it, running it, storing the setup file, installing it, updating it etc. in case of portable app, doesn't even need installation.

Many people, in many cases, do not need Winget to install applications. I am talking about the situation where they do not need it, rather than saying that it is bad.

> Winget doesn't occupy lots of space at all so if reason for uninstalling it is to free up space, it's futile.

I didn't say that.

> Winget is as essential in Window as Apt is in Linux.

But I can still uninstall apt on Linux and just use dpkg.

> PowerShell is a whole different thing, for automation, scripting etc.

I can use PowerShell commands `Add-AppPackage` to install appx-based programs.

---
In a word, the reason why I wrote this issue is because this application can be uninstalled in previous versions of Windows, but it becomes non uninstallable after I reinstall the system.

**Comment by fxzxmicah** [comment]
By the way, I have never used Winget, so I am one of the "many people" I said. Because for me, once I have configured my system, I have all the programs I need, so there is no need to install or uninstall programs.

**Comment by SpecterShell** [comment]
You can use some other softwares such as Geek Uninstaller, DISM++ to uninstall it.
Although it is not suggested to do so since AppInstaller may also work as a dependency for other apps such as Microsoft Store.

**Comment by fxzxmicah** [comment]
> You can use some other softwares such as Geek Uninstaller, DISM++ to uninstall it. Although it is not suggested to do so since AppInstaller may also work as a dependency for other apps such as Microsoft Store.

Not a dependency, I tried in another computer.
And because of this property: `NonRemovable: True`, DISM++ can't uninstall it.

**Comment by fxzxmicah** [comment]
> you thought Winget is used only for Appx packages and I explained it doesn't and it's used to install and update normal every day programs that everyone uses, like .exe and .msi installers.

No, I know exactly what winget is. In my opinion, it is you who don't know the existence of Windows Appinstaller.

> so literally, every person who wants to install something after clean installing Windows, can use Winget to do so, not restricted to Appx packages, even Microsoft store supports many Win32 programs now.

This has nothing to do with my issue. All I see is that you're continually promoting Winget to me.

**Comment by fxzxmicah** [comment]
> it's rather pointless, because I can say 8 billion people use Winget and your statement is invalid, but that's pointless, just like you saying people here or there don't use Winget. nobody set up a poll. 🙂

I don't know why you keep emphasizing this. I'm stating a fact, but it doesn't need proof.

**Comment by fxzxmicah** [comment]
> > > it's rather pointless, because I can say 8 billion people use Winget and your statement is invalid, but that's pointless, just like you saying people here or there don't use Winget. nobody set up a poll. 🙂
> > 
> > 
> > I don't know why you keep emphasizing this. I'm stating a fact, but it doesn't need proof.
> 
> Lol, okay friend, I must've been the one talking about appx package, right right, my bad.
> 
> good luck with your issue and "facts"

I know what Winget is, I know what appinstaller is, and I know the two are bound together. Appinstaller is used to install appx programs, so I mentioned appx. In addition, appinstaller is theoretically required to install appx programs, but in fact PowerShell can also be used to install, so I mentioned PowerShell. And I don't mention traditional Win32 because in addition to using winget, they can also be installed by simply double-clicking the executable, which everyone will do, so I don't mention it.
Finally, your reply here has nothing to do with this issue. I hope you can keep it to the point next time.
Like this:

> ~You can use some other softwares such as Geek Uninstaller, DISM++ to uninstall it.~ Although it is not suggested to do so since AppInstaller may also work as a dependency for other apps such as Microsoft Store.

**Comment by SpecterShell** [comment]
> Not a dependency, I tried in another computer.

At least Microsoft Store is using WinGet to install those new Win32 packages in Microsoft Store (which are uploaded by the publishers, not from winget-pkgs). I don't know what errors will occur if WinGet is uninstalled because I cannot uninstall it either. 😕
![image](https://user-images.githubusercontent.com/56779163/202146627-d6d315e4-a079-4283-80ee-c1495f1735d9.png)

**Comment by fxzxmicah** [comment]
It seems that it does not use winget.exe? I didn't test Win32 in the store.

**Comment by SpecterShell** [comment]
It uses `WindowsPackageManagerServer.exe` instead.
![image](https://user-images.githubusercontent.com/56779163/202160935-ae9a4721-af97-40ad-96b7-752351bae0da.png)

**Comment by fxzxmicah** [comment]
Fine, that's it.

**Comment by salvattore001** [comment]
> why does windowspackagemanagerserver.exe start automatically in background and attempts internet connection. how can i stop it from doing so ?

same thing here. it consumes 25% of CPU and can't end the process. this is crazy and unnaceptable

**Comment by rcompier** [comment]
> why does windowspackagemanagerserver.exe start automatically in background and attempts internet connection. how can i stop it from doing so ? for now i just block it in firewall

I've never seen it run before and suddenly it's there, using up bandwidth and closing the process just restarts a new one, feels more like a infection than being useful to me. If I want something updated I do it myself.
Got to love the random background downloads and updates without notice before your entire gig is sucked dry 👍

**Comment by karelv** [comment]
+1 (same observation as @rcompier)
